### ISSUE TITLE
Use separate front-end like Chrome DevTools app, instead of including own

### ISSUE THREAD

**Issue by auchenberg** [node-inspector/node-inspector#506]
I just released [Chrome DevTools app](https://github.com/auchenberg/chrome-devtools-app), which is a standalone version of Chrome DevTools, independent of Chrome.

![Intro](https://raw.githubusercontent.com/auchenberg/chrome-devtools-app/master/readme/app-intro.png)

I imagine we could make it possible for node-inspector to use [Chrome DevTools app](https://github.com/auchenberg/chrome-devtools-app), as the front-end instead of hosting it's own version?

We could also explore how to broadcast the node-inspector endpoint via mDNS, similar to how [PonyDebugger](https://github.com/square/PonyDebugger) is making endpoints discoverable by the front-end. 

How about that?
##

**Comment by 3y3** [comment]
Are you propose to completely deprecate `frontend` folder in Node Inspector project?
If it's so, then it's impossible, because Inspector's frontend has some specials.

**Comment by bajtos** [comment]
Also keep in mind that the frontend is tightly coupled with the protocol version implemented by the backend. For example, the current Node Inspector version implements protocol from 09/2013. Recent versions of dev-tools frontend may require new commands and/or different response format, depending on the changes made to the protocol in the past.

Having said that, I am open to experimentation as long as the integration with Node Inspector is lightweight and don't add extra maintenance for us.

**Comment by auchenberg** [comment]
@3y3 Yes I'm proposing to have a "generic" DevTool app, that is able to work with multiple targets, ex. chrome, firefox, node-inspector, PonyDebugger, etc.  The point here is that each "back-end provider" shouldn't include their own front-end, but focus on providing a backend that is compliant to the Chrome Remote Debugging protocol.

What are the special needs of the node-inspector front-end? It's my understanding that you are using a pull of DevTools from blink, where you are overriding a few configurations, like toggling certain tabs. We should be able to find a way to avoid these overrides.

@bajtos That's a good point. With the latest pull of DevTools from blink master, I'm seeing a few "unsupported errors" exceptions being thrown, but that should be solvable too.

**Comment by 3y3** [comment]
I would have postponed the issue for at least six months.
Before this discussion I prefer to update frontend, inject DevTools protocol in app, make Node Inspector more extensible, add gui configuration.

If after this steps will be possible to use external frontend, I will be glad to exclude front-end folder and will vote +1 on this issue.

**Comment by bajtos** [comment]
@3y3 I think you misunderstood the intent of this issue. AFAIU, @auchenberg wants to allow both usages: to run the UI shipped inside Node Inspector, but also to use chrome-devtools-app as the front-end. Nobody is asking to remove the front-end folder from Node Inspector, at least not yet.

**Comment by 3y3** [comment]
@bajtos , on my question
> Are you propose to completely deprecate frontend folder in Node Inspector project?

response
> @3y3 Yes I'm proposing to have a "generic" DevTool app, that is able to work with multiple targets, ex. chrome, firefox, node-inspector, PonyDebugger, etc. The point here is that each "back-end provider" shouldn't include their own front-end, but focus on providing a backend that is compliant to the Chrome Remote Debugging protocol.

This misinformed me.

So, looking on some parts of `chrome-devtools-app` I think we can be fully compatible, but it needs to introduce extending API. @auchenberg please look at #492 in folder `front-end-node`. Here are some changes that we need to dinamically inject in `chrome-devtools-app` to be compatible.

**Comment by bajtos** [comment]
Ah, it seems I misunderstood @auchenberg intent. @3y3 you were right in your previous [comment](#issuecomment-69647731). I agree with you that the time to remove front-end from Node Inspector has not come yet.

However, until that time comes, I am personally ok with landing small tweaks that will allow chrome-devtools-app to debug Node apps via Node Inspector backend. That way @auchenberg can explore the concept and refine the implementation, so that when we decide to make the switch, the new solution will be already battle-tested.

**Comment by auchenberg** [comment]
I'm glad to hear you are open to ``Chrome Devtools app``. I'm going to explore the concept more, and dive into the adjustments/overrides you have been making to the Blink Devtools. Thanks for the pointer @3y3. 

My hope is that we, maybe, someday could have a unified DevTool working with node, browsers etc.

**Comment by knownasilya** [comment]
Has there been any movement on this?
